src/signatureVerifier.test.ts

```diff
@@ -0,0 +1,255 @@
+import path from 'path';
+import fs from 'fs';
+import crypto from 'crypto';
+
+import { bufferToHex } from '@zk-email/helpers';
+
+import { SignatureVerifier } from './signatureVerifier.js';
+import { Bytes, Gadgets, UInt32 } from 'o1js';
+import { Bigint2048 } from './rsa.js';
+import {
+  wordToBytes,
+  convertBigIntToByteArray,
+  decompressByteArray,
+} from './utils.js';
+import { BLOCK_SIZES } from './utils.js';
+
+// Path for QR and public key data files.
+const __dirname = path.dirname(new URL(import.meta.url).pathname);
+const qrPath = path.resolve(__dirname, '../src/assets/test.json');
+const certPath = path.resolve(__dirname, '../src/assets/testPublicKey.pem');
+
+const proofsEnabled = false;
+
+describe('Signature Verifier', () => {
+  let qrData: string;
+  let pkData: string;
+  let publicKey: crypto.KeyObject;
+  let testQRData: string;
+  let QRData: bigint;
+
+  let qrDataBytes: Uint8Array<ArrayBufferLike>;
+  let decodedData: Uint8Array<ArrayBufferLike>;
+
+  let signatureBytes: Uint8Array<ArrayBufferLike>;
+  let signedData: Uint8Array<ArrayBufferLike>;
+
+  let publicKeyBigint: Bigint2048;
+  let signatureBigint: Bigint2048;
+
+  beforeAll(async () => {
+    await SignatureVerifier.compile({ proofsEnabled });
+
+    // Get QR data file
+    qrData = fs.readFileSync(qrPath, 'utf8');
+
+    // Get public key data from certificate
+    pkData = fs.readFileSync(certPath, 'utf8');
+
+    // Generate public key from certificate.
+    publicKey = crypto.createPublicKey(pkData);
+
+    // Get the test QR data
+    testQRData = JSON.parse(qrData).testQRData;
+
+    // Convert QR data to bigint
+    QRData = BigInt(testQRData);
+
+    // Parse qr data and convert it to decompressed bytes step by step (using Aadhaar SDK)
+    // Convert QR data bigint to byte array and decompress it
+    qrDataBytes = convertBigIntToByteArray(QRData);
+    decodedData = decompressByteArray(qrDataBytes);
+
+    // Split signature and signed data. Signature in aadhaar QR data is the last 256 bytes.
+    signatureBytes = decodedData.slice(
+      decodedData.length - 256,
+      decodedData.length
+    );
+
+    signedData = decodedData.slice(0, decodedData.length - 256);
+
+    // Convert pubkey from byte array to BigInt type.
+    publicKeyBigint = Bigint2048.from(
+      BigInt(
+        '0x' +
+          bufferToHex(
+            Buffer.from(
+              publicKey.export({ format: 'jwk' }).n as string,
+              'base64url'
+            )
+          )
+      )
+    );
+
+    // Convert signature from byte array to BigInt type.
+    signatureBigint = Bigint2048.from(
+      BigInt('0x' + bufferToHex(Buffer.from(signatureBytes)).toString())
+    );
+  });
+
+  describe('Partial hashing computations', () => {
+    it.skip('should compute partial hashing with 9 byte blocks of size 128 bytes.', async () => {
```

Comment:
You can automatize the activation of this test in the case when `proofsEnabled=true` as follows:
```suggestion
    (!proofsEnabled ? it.skip : it)('should compute partial hashing with 9 byte blocks of size 128 bytes.', async () => {
```

Reply:
Updated file: [signatureVerifier.test.ts](https://github.com/mina-builders-team/anon-aadhaar-o1js/pull/2/commits/2fb75109a49358ce2509257b22409c41e56820d4#diff-fb933622ad795786da1fecc00bf398e042c26ce68697964f2574daf55e670d80R90-R94)